Apps/Src/SendTritium.c

```diff
@@ -280,10 +285,22 @@ static void readInputs(){
  * @returns integer value from out_min to out_max
 */
 static uint8_t map(uint8_t input, uint8_t in_min, uint8_t in_max, uint8_t out_min, uint8_t out_max){
-    if(in_max >= in_min) in_max = in_min;
-    if(input <= in_min) return out_min;
-    else if(input >= in_max) return out_max;
-    else return ((out_max - in_max) * (input - in_min))/(out_min - in_min) + out_min;
+    if(in_min >= in_max) in_max = in_min;   // The minimum of the input range should never be greater than the maximum of the input range
+    
+    if(input <= in_min){
+        // Lower bound the input to the minimum possible output
+        return out_min;
+    } else if(input >= in_max){
+        // Upper bound the input to the maximum output
+        return out_max;
+    } else{
+        // Linear mapping between ranges
+        uint8_t offset_in = input - in_min;  // If input went from A -> B, it now goes from 0 -> B-A
+        uint8_t in_range = in_max - in_min;   // Input range
+        uint8_t out_range = out_max - out_min;    // Output range
+        uint8_t offset_out = out_min;
+        return (offset_in * out_range)/in_range + offset_out;   // slope = out_range/in_range. y=mx+b so output=slope*offset_in+offset_out
```

Comment:
Overall looks fine. Just have one question. If `(offset_in * out_range)` is less than `in_range`, the first term of the last line will zero out, and only offset_out will be returned. Is that desired behavior?

Reply:
The reason I ask is that if `out_range` is sufficiently smaller than `in_range`, it would return the same output for a range of input values, which I just want to confirm is what it is meant to do.